### PR TITLE
Fix compiling error and some clippy warnings.

### PR DESCRIPTION
**Pull request by necokeine**

### CODE REVIEW

File: benches/rate_calculation.rs [hunk 1]
```diff
@@ -139,8 +139,8 @@ fn criterion_benchmark(c: &mut Criterion) {
     // Define timestep
     world.insert(Timestep { delta: 1.0e-6 });
 
-    let vel_dist = Normal::new(0.0, 0.22);
-    let pos_dist = Normal::new(0.0, 1.2e-4);
+    let vel_dist = Normal::new(0.0, 0.22).unwrap();
```
Comment (necokeine) [inline benches/rate_calculation.rs:142]: Here, a compile error.

File: examples/quadrupole_trap.rs [hunk 2]
```diff
@@ -88,7 +88,7 @@ fn main() {
     }
 
     // Define timestep
-    world.add_resource(Timestep { delta: 1.0e-5 });
+    world.insert(Timestep { delta: 1.0e-5 });
```
Comment (necokeine) [inline examples/quadrupole_trap.rs:91]: Replace a deprecated interface.

File: benches/rate_calculation.rs (hunk 1 above)
Comment (ElliotB256) [reply benches/rate_calculation.rs:142]: Thanks! the benches are a bit unfinished for now, I will create a new issue to add more benchmarks to the bench suite.

**Comment by ElliotB256** [comment]
After merging the other PR into master, this one flagged up a few different conflicts. They look to be mostly inconcistencies on tabs vs spaces, from what I could tell - could you verify if that is correct? thanks!

**Comment by necokeine** [comment]
I suppose you want to rebase this PR onto the new master head, I will remove the last Merge branch commit and rebase my PR onto the newest master branch.

**Comment by necokeine** [comment]
Hi @ElliotB256 I replaced all tab in this project into 4 spaces, and it looks better now.
If there is still some new conflict, or any other required changes, please comment here and I will do those changes.

**Comment by ElliotB256** [comment]
cheers, starting the CI now

**Comment by necokeine** [comment]
Could you help me to merge this PR? it seems I don't have permission to do so.

**Review by ElliotB256 — APPROVED** [review]

**Comment by ElliotB256** [comment]
I'ved merged it now, thanks for submitting :) I started the CI before I left yesterday and only just came back to it

**Pull request merged**